File: src/02.string/03.count/06.451.go
package count

import (
	"sort"
)

func FrequencySort(s string) string {
	/**
	mp := make(map[byte]int)

	for i := range s {
		mp[s[i]]++
	}

	type arrItem struct {
		ch    byte
		count int
	}

	arr := make([]arrItem, 0, len(mp))

	for k, v := range mp {
		arr = append(arr, arrItem{
			ch:    k,
			count: v,
		})
	}

	sort.Slice(arr, func(i, j int) bool {
		return arr[j].count < arr[i].count
	})

	res := make([]byte, 0, len(s))
	for _, v := range arr {
		res = append(res, bytes.Repeat([]byte{v.ch}, v.count)...)
	}

	return string(res)
	*/

	type arrItem struct {
		ch    string
		count int
	}

	arr := make([]arrItem, 128)

	for i := range s {
		arr[s[i]].ch += string(s[i])
		arr[s[i]].count++
	}

	sort.Slice(arr, func(i, j int) bool {
		return arr[i].count > arr[j].count
	})

	res := ""
	for _, v := range arr {
		res += v.ch
	}

	return res
}

File: src/02.string/03.count/06.451.ts
export function frequencySort(s: string): string {
    const arrLen = 128;
    const arr: { code: string; count: number }[] = new Array(arrLen)
        .fill(0)
        .map(() => ({
            code: "",
            count: 0,
        }));

    for (let i = 0; i < s.length; i++) {
        const index = s[i].charCodeAt(0);
        arr[index].code += s[i];
        arr[index].count++;
    }

    arr.sort((a, b) => {
        return b.count - a.count;
    });

    let res = "";
    for (let i = 0; i < arrLen; i++) {
        if (arr[i].count > 0) {
            res += arr[i].code;
        } else {
            return res;
        }
    }

    return res;
}
